File: 160-intersection-of-two-linked-lists/160-intersection-of-two-linked-lists.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    def getIntersectionNode(self, headA: ListNode, headB: ListNode) -> Optional[ListNode]:
        if (headA == None or  headB == None):
            return None
        
        p1 = headA
        p2 = headB
        
        while (p1 and p2 and p1 != p2 ):
            p1 = p1.next
            p2 = p2.next
            
            if p1 == p2:
                return p1
            
            if p1 == None:
                p1 = headB
                
            if p2 == None:
                p2 = headA
                
            
        return p1

File: 160-intersection-of-two-linked-lists/160-intersection-of-two-linked-lists.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) {
 *         val = x;
 *         next = null;
 *     }
 * }
 */
public class Solution {
    public ListNode getIntersectionNode(ListNode headA, ListNode headB) {
        ListNode travA = headA;
        ListNode travB = headB;
        
        while (travA!= null || travB!= null){
            if (travA == null){
                travA = headB;
            }
            
            if (travB == null){
                travB = headA;
            }
            
            if (travA == travB ){
                return travA;
            }

            travA = travA.next;
            travB = travB.next;
            
        }
        
        return null;
    }
}